src/Security/Authentication/OpenIdConnect/src/OpenIdConnectConfigureOptions.cs

```diff
@@ -0,0 +1,136 @@
+    /// </summary>
+    /// <param name="configurationProvider">An <see cref="IAuthenticationConfigurationProvider"/> instance.</param>
+    public OpenIdConnectConfigureOptions(IAuthenticationConfigurationProvider configurationProvider)
+    {
+        _authenticationConfigurationProvider = configurationProvider;
+    }
+
+    /// <inheritdoc />
+    public void Configure(string? name, OpenIdConnectOptions options)
+    {
+        if (string.IsNullOrEmpty(name))
+        {
+            return;
+        }
+
+        var configSection = _authenticationConfigurationProvider.GetSchemeConfiguration(name);
+
+        if (configSection is null || !configSection.GetChildren().Any())
+        {
+            return;
+        }
+
+        options.AccessDeniedPath = new PathString(configSection[nameof(options.AccessDeniedPath)]);
+        options.Authority = configSection[nameof(options.Authority)];
+        options.AutomaticRefreshInterval = StringHelpers.ParseValueOrDefault(options.AutomaticRefreshInterval, configSection[nameof(options.AutomaticRefreshInterval)], _invariantTimeSpanParse);
+        options.BackchannelTimeout = StringHelpers.ParseValueOrDefault(options.BackchannelTimeout, configSection[nameof(options.BackchannelTimeout)], _invariantTimeSpanParse);
+        options.CallbackPath = new PathString(configSection[nameof(options.CallbackPath)] ?? options.CallbackPath);
+        options.ClaimsIssuer = configSection[nameof(options.ClaimsIssuer)];
+        options.ClientId = configSection[nameof(options.ClientId)];
+        options.ClientSecret = configSection[nameof(options.ClientSecret)];
+        options.CorrelationCookie = GetCookieFromConfig(configSection.GetSection(nameof(options.CorrelationCookie)), options.CorrelationCookie);
+        options.DisableTelemetry = StringHelpers.ParseValueOrDefault(options.DisableTelemetry, configSection[nameof(options.DisableTelemetry)], bool.Parse);
+        options.ForwardAuthenticate = configSection[nameof(options.ForwardAuthenticate)];
+        options.ForwardChallenge = configSection[nameof(options.ForwardChallenge)];
+        options.ForwardDefault = configSection[nameof(options.ForwardDefault)];
+        options.ForwardForbid = configSection[nameof(options.ForwardForbid)];
+        options.ForwardSignIn = configSection[nameof(options.ForwardSignIn)];
+        options.ForwardSignOut = configSection[nameof(options.ForwardSignOut)];
+        options.GetClaimsFromUserInfoEndpoint = StringHelpers.ParseValueOrDefault(options.GetClaimsFromUserInfoEndpoint, configSection[nameof(options.GetClaimsFromUserInfoEndpoint)], bool.Parse);
+        options.MapInboundClaims = StringHelpers.ParseValueOrDefault(options.MapInboundClaims, configSection[nameof(options.MapInboundClaims)], bool.Parse);
+        options.MaxAge = options.MaxAge is TimeSpan maxAge ? StringHelpers.ParseValueOrDefault(maxAge, configSection[nameof(options.MaxAge)], _invariantTimeSpanParse) : options.MaxAge;
```

Comment:
This doesn't look right. It seems like if MaxAge is not set then the config will be ignored? Shouldn't ParseValueOrDefault work fine regardless of if the initial MaxAge was null?

Reply:
💀 Fixed.